Provide modules.pillar.obfuscate and modules.pillar.ls

gracinet opened the pull request:
This fulfills #19329. As explained there, 'ls' is not that useful, but still including it for completeness.
Also first attempt at writing a unit test within salt.

Commit:
Provide modules.pillar.obfuscate and modules.pillar.ls

This fulfills #19329. As explained there, 'ls' is not that useful, but
still including it for completeness.
Also first attempt at writing a unit test within salt.

rallytime commented:
@gracinet The pylint errors are not related, and most of the test failures are also unrelated except for one: `integration.modules.sysmod.SysModuleTest.test_valid_docs`. Basically you just need to add a CLI example to `pillar.obfuscate_inner` and that should be cleared up.

gracinet commented:
@rallytime thanks for the tip. The thing is, this `obfuscate_inner()` is not meant for the CLI, but to encapsulate the recursion, and be easy to unit test (since it's actually pure Python without any reference to the Salt environment). What's the proper way to exclude it ?

rallytime commented:
Oh! So this is just used by other functions within the file? Just throw an underscore in front of the function. So something like:
```
def _obfuscate_inner(var)
```

Commit:
Proper versionadded tags

Commit:
Mark function as private

With the leading underscore, _obfuscate_inner should be excluded from the CLI
(and the related lint tests)

gracinet commented:
It seems that integration.modules.sysmod.SysModuleTest.test_valid_docs doesn't complain anymore.

rallytime commented:
Awesome! Thanks for writing some tests here, too!

rallytime commented on tests/unit/modules/pillar_test.py:
```diff
@@ -0,0 +1,58 @@
+
+pillar_value_1 = dict(a=1, b='very secret')
+
+
```
This line needs a:
```
@skipIf(NO_MOCK, NO_MOCK_REASON)
```

rallytime replied:
Oh, actually I see you have that line above the functions needing them. Usually we just put that at the top of the file, but I think this works for now. Thoughts @s0undt3ch?

s0undt3ch replied:
@rallytime if all tests were to be skipped, then the `skipIf`decorator should be added to the class(not mandatory but preferable), but since only some tests are to be skipped, then decorating the methods is best.

gracinet commented:
wonderful, @thatch45, thanks !

thatch45 commented:
Thank you for the additions @gracinet !
